### DISCUSSION TITLE
Can't get opentelemetry logs in skywalking in k8s

### DISCUSSION THREAD

**Discussion by tvildo** [apache/skywalking#12050] [Q&A]
Hi, I deployed skywalking [9.7.0](https://github.com/apache/skywalking/releases/tag/v9.7.0) in kubernetes, I use elasticsearch as database, based on skywalking logs it seems it is up and running. But In elastic tables are empty. No logs are ingested by skywalking but they are sent from opentelemetry pod and I see no error as well. 

Am I missing something, What can I do to make it work ?

Skywalking config:
```
{
	"SW_CLUSTER": "kubernetes",
	"SW_CLUSTER_K8S_LABEL": "app=skywalking,component=oap",
	"SW_CLUSTER_K8S_NAMESPACE": "dev-david-monitoring",
	"SW_ES_PASSWORD": "Q9XKEp14wwwwwDeAvvOQJTEST",
	"SW_ES_USER": "admin",
	"SW_OTEL_RECEIVER": "default",
	"SW_OTEL_RECEIVER_ENABLED_HANDLERS": "otlp-metrics,otlp-logs,otlp-traces",
	"SW_STORAGE": "elasticsearch",
	"SW_STORAGE_ES_CLUSTER_NODES": "elasticsearch-es-http.dev-david-monitoring.svc:9200",
	"SW_STORAGE_ES_HTTP_PROTOCOL": "https",
	"SW_STORAGE_ES_INDEX_REPLICAS_NUMBER": "0",
	"SW_STORAGE_ES_INDEX_SHARDS_NUMBER": "1",
	"SW_STORAGE_ES_SSL_JKS_PASS": "jM22ORGwojwjjBllYS2lTEST",
	"SW_STORAGE_ES_SSL_JKS_PATH": "/jks/certs.jks"
}
```
Following is log from skywalking:
[skywalking_log.txt](https://github.com/apache/skywalking/files/14748141/skywalking_log.txt)

This is what is in elastic: 
![image](https://github.com/apache/skywalking/assets/8401382/e87d66b6-eee6-47fc-8216-6aa5b8be8e4a)

This is my opentelemetry helm chart config related to skywalking:
```
config:
  exporters:
    otlp:
      endpoint: skywalking.dev-david-monitoring.svc.cluster.local:11800
      tls:
        insecure: true
    debug: {}
    logging: {}
  extensions:
    health_check:
      endpoint: ${env:MY_POD_IP}:13133
    memory_ballast: {}
  processors:
    batch: {}
    memory_limiter: null
  receivers:
    jaeger:
      protocols:
        grpc:
          endpoint: ${env:MY_POD_IP}:14250
        thrift_http:
          endpoint: ${env:MY_POD_IP}:14268
        thrift_compact:
          endpoint: ${env:MY_POD_IP}:6831
    otlp:
      protocols:
        grpc:
          endpoint: ${env:MY_POD_IP}:4317
        http:
          endpoint: ${env:MY_POD_IP}:4318
    prometheus:
      config:
        scrape_configs:
          - job_name: opentelemetry-collector
            scrape_interval: 10s
            static_configs:
              - targets:
                  - ${env:MY_POD_IP}:8888
    zipkin:
      endpoint: ${env:MY_POD_IP}:9411
  service:
    telemetry:
      metrics:
        address: ${env:MY_POD_IP}:8888
    extensions:
      - health_check
      - memory_ballast
    pipelines:
      logs:
        exporters:
          - debug
        processors:
          - memory_limiter
          - batch
        receivers:
          - otlp
      metrics:
        exporters:
          - debug
          - otlp
        processors:
          - memory_limiter
          - batch
        receivers:
          - otlp
          - prometheus
      traces:
        exporters:
          - debug
          - otlp
        processors:
          - memory_limiter
          - batch
        receivers:
          - otlp
          - jaeger
          - zipkin
```

**Comment by wu-sheng** [comment]
You could follow this doc, https://skywalking.apache.org/docs/main/next/en/setup/backend/log-otlp/

I assume `service.name` is missed in your raw logs.

**Reply by tvildo** [reply]
Hi, I followed that guide and got the same result. I am getting logs in skywalking but index in elasticsearch is empty again.
I attached remote debugger to skywalking server and I am able to confirm that is is connected to elasticsearch and also It is receiving metrics from opentelemetry collector which you can see below. 
<img width="1492" alt="Screenshot 2024-04-19 at 23 47 27" src="https://github.com/apache/skywalking/assets/8401382/9897fd51-48df-4df1-9bb1-15cc9f8d62c4">

I can't see how those metrics are persisted in elastic database. Can someone please suggest where I can set breakpoint to investigate my issue further?

**Reply by wu-sheng** [reply]
How about service name? As you have debugging, please continue to follow the codes.

**Reply by kellis5137** [reply]
Any luck with this? I can get logs coming in fine, but no metrics.

**Reply by tvildo** [reply]
No. I am still trying to make it work. Which version of skywalking do you use?

**Reply by wu-sheng** [reply]
https://github.com/apache/skywalking/discussions/12168

We addressed a bug in the demo, which related to logs, https://github.com/apache/skywalking/discussions/12168

But I am not sure whether it is same as yours.

**Reply by kellis5137** [reply]
9.7?! I went back to 9.6, but same thing.
